src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -200,25 +200,35 @@
+ (fn [{:keys [db]} [{:keys [token-symbol token stack-id start-flow? owners]} entry-point]]
    ;; `token` is a map extracted from the sender, but in the wallet home page we don't know the
    ;; sender yet, so we only provide the `token-symbol`, later in
    ;; `:wallet/select-from-account` the `token` key will be set.
-   (let [{token-networks :networks}                token
-         receiver-networks                         (get-in db [:wallet :ui :send :receiver-networks])
-         token-networks-ids                        (mapv #(:chain-id %) token-networks)
-         token-not-supported-in-receiver-networks? (not-any? (set receiver-networks)
-                                                             token-networks-ids)]
-     (when (or token token-symbol)
+   (let [{:keys [networks]}  token
+         receiver-networks   (get-in db [:wallet :ui :send :receiver-networks])
+         token-networks-ids  (map :chain-id networks)
+         unsupported-token?  (not-any? (set receiver-networks) token-networks-ids)
+         unique-owner        (when (= (count owners) 1)
+                               (first owners))
+         unique-owner-tokens (get-in db [:wallet :accounts unique-owner :tokens])
+         token-data          (or token
+                                 (when (and token-symbol unique-owner)
+                                   (some #(when (= (:symbol %) token-symbol) %)
+                                         unique-owner-tokens)))]
+     (when (or token-data token-symbol)
        {:db (cond-> db
-              :always      (update-in [:wallet :ui :send] dissoc :collectible)
-              :always      (assoc-in
-                            [:wallet :ui :send :token-not-supported-in-receiver-networks?]
-                            token-not-supported-in-receiver-networks?)
-              token        (assoc-in [:wallet :ui :send :token] token)
-              token        (assoc-in [:wallet :ui :send :token-display-name]
-                            (:symbol token))
-              token-symbol (assoc-in [:wallet :ui :send :token-symbol] token-symbol))
+              :always      (update-in [:wallet :ui :send]
+                                      #(-> %
+                                           (dissoc :collectible)
+                                           (assoc :token-not-supported-in-receiver-networks?
+                                                  unsupported-token?)))
+              token-symbol (assoc-in [:wallet :ui :send :token-symbol] token-symbol)
+              token-data   (update-in [:wallet :ui :send]
+                                      #(assoc %
+                                              :token              token-data
+                                              :token-display-name (:symbol token-data)))
+              unique-owner (assoc-in [:wallet :current-viewing-account-address] unique-owner)
+              entry-point  (assoc-in [:wallet :ui :send :entry-point] entry-point))
```

Comment:
Entry point is like the screen the user started the wizard flow on?
Can't say I'm a fan of coupling this ui logic to the wizard logic. 
Is there no other way to handle this? Do we need to do it?

Reply:
Yes, that's exactly what `:entry-point` does.

We need to know where the flow started because the back button will clean different things according to the `:entry-point`.

E.g.
If we start on wallet home and there's only one account owning an asset, we set the `current-viewing-account` as that account and we remove it when we navigate back.

But if we start the flow from inside the account, we can't remove `current-viewing-account` since this leads to an empty state of the view because the user was watching its account.

As I mentioned in a self-review comment, we shouldn't have used `current-viewing-account` (an UI thing) to determine the sender, but a refactor at this point is hard to perform.

Reply:
Would the entry-point logic be necessary if we had a navigation graph in our re-frame database, where we could retrieve the navigation stacks/history? I'm considering whether this could be a valid use case to take into account when refactoring the navigation logic in the future. I recall trying to implement this in the past (#19098), but it introduced extra complexity that wasn't worth adding at the time. However, it might be useful now if I didn't misunderstand the issue.

Reply:
Yes, I think we could skip the entry point if we know the current navigation stack.

This is a point to consider when we change the navigation library, we can store the navigation in re-frame. I've done it in the past and it's not complicated in React Navigation.
-  https://github.com/status-im/status-mobile/issues/20471